File: selectionSort.java
/**
* @author 	Luca Cortinovis	luca1cortinovis@gmail.com
* @version 	1.0
* @since	24/5/2021
*
* You can find me here:
* GitHub	Luca Cortinovis(NeptuneLuke)
*/

import java.util.Scanner;
import java.util.Arrays;

public class selectionSort {


	public static void main(String[] args) {
		
		Scanner keyboard = new Scanner(System.in);
		
		int myArray[] = new int[5];
		for(int i=0; i<myArray.length; i++) {
			System.out.println("Insert number: ");
			myArray[i] = keyboard.nextInt();
		}
		System.out.println();
		
		selectionSort(myArray);
		
		System.out.println(Arrays.toString(myArray));
	}

	
	public static void selectionSort(int myArray[]) {
		
		for(int i=0; i<myArray.length-1; i++) {
			
			int min = i;
			for(int j=i+1; j<myArray.length; j++) {
				
				if(myArray[j] < myArray[min])
					min = j;
			}
			
			swap(myArray,i,min);
		}
		
	}
	
	public static void swap(int myArray[], int a, int b) {
		
		int temp = myArray[a];
	    myArray[a] = myArray[b];
	    myArray[b] = temp;
	}
	
	
}

File: sorting-algorithms/selection_sort.cpp
/**
* @author 	Luca Cortinovis	luca1cortinovis@gmail.com
*
* You can find me here:
* GitHub	Luca Cortinovis(NeptuneLuke)
*/

#include <iostream>
#include <vector>


using
std::vector,
std::cout,
std::cin;


void selection_sort(vector<int>& myVector);
void swap(vector<int>& myVector, int a, int b);


int main(void) {

	vector<int> myVector(5);
	int number;
	for (int i = 0; i < myVector.size(); ++i) {
		cout << "Insert number: ";
		cin >> number;
		myVector[i] = number;
		cout << "\n";
	}


	selection_sort(myVector);

	for (int i = 0; i < myVector.size(); ++i) {
		cout << myVector[i] << "\n";
	}

	return 0;
}


void selection_sort(vector<int>& myVector) {

	for (int i = 0; i < myVector.size() - 1; ++i) {

		int min = i;
		for (int j = i + 1; j < myVector.size(); ++j) {

			if (myVector[j] < myVector[min]) {
				min = j;
			}
		}

		swap(myVector, i, min);
	}

}

void swap(vector<int> &myVector, int a, int b) {

	int temp = myVector[a];
	myVector[a] = myVector[b];
	myVector[b] = temp;
}